### PR TITLE
Cleanup code in ssl package.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

There were some warnings for the code in the ssl package.

Modifications:

- Remove not needed else blocks
- Use correctly base class for static usage
- Replace String.length() == 0 with String.isEmpty()
- Remove unused code

Result:

Less warnings and cleaner code.

### CODE REVIEW

File: handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java [hunk 1]
```diff
@@ -902,7 +898,8 @@ public final SSLEngineResult unwrap(
                                 if (!dst.hasRemaining()) {
                                     // Move to the next dst buffer as this one is full.
                                     continue;
-                                } else if (packetLength == 0) {
+                                }
+                                if (packetLength == 0) {
```
Comment (netkins) [inline handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java]: MAJOR Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS134)

File: handler/src/main/java/io/netty/handler/ssl/OpenSsl.java [hunk 2]
```diff
@@ -125,14 +125,14 @@
                     try {
                         for (String c: SSL.getCiphers(ssl)) {
                             // Filter out bad input.
-                            if (c == null || c.length() == 0 || availableOpenSslCipherSuites.contains(c)) {
+                            if (c == null || c.isEmpty() || availableOpenSslCipherSuites.contains(c)) {
```
Comment (netkins) [inline handler/src/main/java/io/netty/handler/ssl/OpenSsl.java]: MAJOR Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS134)

File: handler/src/main/java/io/netty/handler/ssl/SslHandler.java [hunk 3]
```diff
@@ -34,6 +34,7 @@
 import io.netty.channel.PendingWriteQueue;
 import io.netty.handler.codec.ByteToMessageDecoder;
 import io.netty.handler.codec.UnsupportedMessageTypeException;
+import io.netty.util.ReferenceCounted;
```
Comment (netkins) [inline handler/src/main/java/io/netty/handler/ssl/SslHandler.java]: MINOR Remove this unused import 'io.netty.util.ReferenceCounted'. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

Comment (Scottmitch) [inline handler/src/main/java/io/netty/handler/ssl/SslHandler.java]: +1

Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/ssl/SslHandler.java]: fixed

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (f7c8cf9) and 4.0 (f2f707c)

**Pull request closed without merging**